x-pack/metricbeat/module/azure/monitor/azure_monitor_service.go

```diff
@@ -0,0 +1,91 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package monitor
+
+import (
+	"context"
+	"fmt"
+	"github.com/Azure/azure-sdk-for-go/services/preview/monitor/mgmt/2019-06-01/insights"
+	"github.com/Azure/azure-sdk-for-go/services/resources/mgmt/2019-03-01/resources"
+	"github.com/Azure/go-autorest/autorest/azure/auth"
+	"github.com/elastic/beats/libbeat/logp"
+)
+
+type AzureMonitorService struct {
+	metricsClient          *insights.MetricsClient
+	metricDefinitionClient *insights.MetricDefinitionsClient
+	resourceClient         *resources.Client
+	log                    *logp.Logger
+}
+
+// Init instantiates the an Azure monitoring service
+func Init(clientID string, clientSecret string, tenantID string, subscriptionID string) (*AzureMonitorService, error) {
```

Comment:
I would call this `New` or `NewAzureMonitorService`. Otherwise can be confused with `init` what I first thought it is.